### DISCUSSION TITLE
I can't get huge texts

### DISCUSSION THREAD

**Discussion by Fenish** [cefsharp/CefSharp#3480] [Q&A]
when i try to get huge texts with javascript, it cames null.
for example i am httprequesting to website and im getting result with js, when i check console, its perfect but when i debug with msgbox.show, it comes nothing.

**Os**: windows 7
**Cefsharp version**: 88.2.90.0
**Platform**: winforms
**Cpu**: x86

Code
```
...
.....
         mainform.browser.ExecuteScriptAsync("var obj = $.get('requested_website');");
         JavascriptResponse result = JsScriptResult("obj['responseText']", 1000);
         MessageBox.Show("Result:" + result.Result.ToString() + "\nMessage: " + result.Message);


        private JavascriptResponse JsScriptResult(string script, int timeout)
        {
            IFrame frame =  mainform.browser.GetMainFrame();
            Task<JavascriptResponse> Task = frame.EvaluateScriptAsync(script, timeout: TimeSpan.FromMilliseconds(timeout));
            Task.Wait();
            return Task.Result;
        }
```

but when i manually type **obj['responseText']** to console, it comes perfect.
And also when i try to do this, it comes perfectly

**Code 2**
```
         mainform.browser.ExecuteScriptAsync("var dadada = '3131' ");
         JavascriptResponse result = JsScriptResult("obj['responseText']", 1000);
         MessageBox.Show("Result:" + result.Result.ToString() + "\nMessage: " + result.Message);

```

How can i get result of request webpages.
**Original obj['responseText']** is kinda like that 
```
<xml>
   <item id="69" name="blabalbal">
</xml>
```

**Comment by amaitland** [comment]
Use http://cefsharp.github.io/api/89.0.x/html/Overload_CefSharp_WebBrowserExtensions_EvaluateScriptAsPromiseAsync.htm and wrap your Ajax call in a promise. See https://github.com/cefsharp/CefSharp/pull/3251 for further details.
